Implement zoneless configuration

MRinaldi9 opened the pull request:
#5 Closing this issue

shairez commented on src/setup.ts:
```diff
@@ -1,12 +1,18 @@
 } from "@angular/platform-browser/testing";
 
-export function setupAngularTestEnvironment(): void {
+export function setupAngularTestEnvironment(isZoneless = true): void {
```
I think it's better to have an "options" object with a "zoneless" boolean instead of a direct variable, makes the call more readable later

shairez commented on src/setup-zoneless.ts:
```diff
@@ -0,0 +1,7 @@
+import "@analogjs/vitest-angular/setup-snapshots";
+import { setupAngularTestEnvironment } from "./setup";
+
+globalThis.__ZONELESS__ = true;
```
why is this required? just for the tests later? or for other reasons?

MRinaldi9 replied:
yeah, i opted for this for skip the tests if not in zoneless, but i think with the configuration on vitest we can solve the problem.

MRinaldi9 commented:
i had changed the tsconfig because in v4 vitest there is no need to set the types for the providers and i added the globals so the tests have less "imports code".

shairez commented:
Thanks for the fixes @MRinaldi9 !

@yjaaidi wanna review before we merge?

yjaaidi commented:
Hi @MRinaldi9, first of all thank you for the contribution.

The thing is that the whole TestBed setup _(including zoneful/zoneless setup)_ is already performed by Angular's `unit-test` builder and `@analog/vitest-angular`.

We still have to discuss this with @shairez but I think that we should reduce the surface of this package to providing a  `render` function and avoid overlapping with Angular CLI's and Analog.

What do you think Shai?
